### PR TITLE
feat(security): Integrate device services with consul tokens

### PR DESCRIPTION
**Pull request by jim-wang-intel**
Make file refactored and changes on some security yaml files
- Add a shell script to handle the generation of secure version of device-related service docker-compose files dynamically based on the Makefile options
- Remove unnecessary `asc-x-secure` yaml files as they are also generated on the fly when it is in security mode

Closes: #62

Signed-off-by: Jim Wang <yutsung.jim.wang@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/master/.github/CONTRIBUTING.md.

## What is the current behavior?

Currently, there is no device related service running in security mode with Consul token for docker-compose files.

## What is the new behavior?
Added integration with device-related services for running in security mode with Consul token for docker-compose files.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

## New Imports
Are there any new imports or modules? If so, what are they used for and why?

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Review by lenny-intel — COMMENTED** [review]
This is way more complex and beyond what I was recommending. Too much duplication of what is in the makefile, require multiple files to be edited when adding a new device service.
I just wanted a script that took the template and generated a extension file for the specified app/device service and returned the filename to be added to the list of compose files.

**Example of use for asc-http from makefile:**
```
ifeq (asc-http, $(filter asc-http,$(ARGS)))
	# always add the base
	COMPOSE_FILES:=$(COMPOSE_FILES) -f add-asc-http-export.yml
	ifneq (no-secty, $(filter no-secty,$(ARGS)))
		ifeq ($(TOKEN_LIST),"")
			TOKEN_LIST:=app-http-export
		else
			TOKEN_LIST:=$(TOKEN_LIST),app-http-export
		endif
		# when no security mode (no-secty) not explicitly specified,
		# then we also need to add the secure version on top of base yml by default.
               extension-file = gen-extension app-service-http-export app-http-export app-service-configurable
		COMPOSE_FILES:=$(COMPOSE_FILES) -f ${extension-file}
	endif
endif
```
In this example it would generate the file gen_staging/add-app-service-http-export-secure.yml

**Comment by jim-wang-intel** [comment]
> This is way more complex and beyond what I was recommending. Too much duplication of what is in the makefile, require multiple files to be edited when adding a new device service.
> I just wanted a script that took the template and generated a extension file for the specified app/device service and returned the filename to be added to the list of compose files.
> 
> **Example of use for asc-http from makefile:**
> 
> ```
> ifeq (asc-http, $(filter asc-http,$(ARGS)))
> 	# always add the base
> 	COMPOSE_FILES:=$(COMPOSE_FILES) -f add-asc-http-export.yml
> 	ifneq (no-secty, $(filter no-secty,$(ARGS)))
> 		ifeq ($(TOKEN_LIST),"")
> 			TOKEN_LIST:=app-http-export
> 		else
> 			TOKEN_LIST:=$(TOKEN_LIST),app-http-export
> 		endif
> 		# when no security mode (no-secty) not explicitly specified,
> 		# then we also need to add the secure version on top of base yml by default.
>                extension-file:= $(shell gen-extension app-service-http-export app-http-export app-service-configurable)
> 		COMPOSE_FILES:=$(COMPOSE_FILES) -f ${extension-file}
> 	endif
> endif
> ```
> 
> In this example it would generate the file gen_staging/add-app-service-http-export-secure.yml

ok, will change it to call the script on each option section; Thanks!

File: compose-builder/gen_compose_ext.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
```
Comment (lenny-intel) [inline compose-builder/gen_compose_ext.sh:1]: How about changing the name slightly to be more explicit that it is for secure extension generation? Like `gen_secure_compose_ext.sh`

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Looks Great, just one minor name change.

File: compose-builder/gen_compose_ext.sh (hunk 1 above)
Comment (jim-wang-intel) [reply compose-builder/gen_compose_ext.sh:1]: sure

Comment (jim-wang-intel) [reply compose-builder/gen_compose_ext.sh]: renamed

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**